File: core-arch/src/main/java/com/huyingbao/core/lifecycle/FragmentLifecycleObserver.java
package com.huyingbao.core.lifecycle;

import android.arch.lifecycle.Lifecycle;
import android.arch.lifecycle.LifecycleObserver;
import android.arch.lifecycle.OnLifecycleEvent;
import android.support.v4.app.Fragment;

import com.huyingbao.core.store.RxStore;
import com.huyingbao.core.view.RxFluxView;

import java.util.List;

/**
 * Created by liujunfeng on 2017/12/13.
 */
public class FragmentLifecycleObserver implements LifecycleObserver {
    private final Fragment mFragment;

    public FragmentLifecycleObserver(Fragment fragment) {
        this.mFragment = fragment;
    }

    /**
     * fragment创建成功之后调用,
     * 若fragment是RxViewDispatch的子类,
     * 获取需要关联的RxStoreList
     * 将RxStoreList同fragment生命周期关联
     */
    @OnLifecycleEvent(Lifecycle.Event.ON_CREATE)
    void onCreate() {
        if (mFragment instanceof RxFluxView) {
            List<RxStore> rxStoreList = ((RxFluxView) mFragment).getLifecycleRxStoreList();
            if (rxStoreList != null && rxStoreList.size() > 0)
                for (RxStore rxStore : rxStoreList)
                    mFragment.getLifecycle().addObserver(rxStore);
        }
    }
}

File: core-arch/src/main/java/com/huyingbao/core/arch/lifecycle/FragmentLifecycleObserver.kt
package com.huyingbao.core.arch.lifecycle

import androidx.fragment.app.Fragment
import androidx.lifecycle.Lifecycle
import androidx.lifecycle.LifecycleObserver
import androidx.lifecycle.OnLifecycleEvent
import com.huyingbao.core.arch.store.FragmentStore
import com.huyingbao.core.arch.view.FluxView

/**
 * Fragment生命周期观察者，将Fragment持有的[com.huyingbao.core.arch.store.Store]关联其生命周期。
 *
 * Created by liujunfeng on 2019/1/1.
 */
class FragmentLifecycleObserver(
        private val fragment: Fragment
) : LifecycleObserver {
    /**
     * 在onAttach()完成依赖注入之后调用
     */
    @OnLifecycleEvent(Lifecycle.Event.ON_CREATE)
    fun onCreate() {
        if (fragment is FluxView<*>) {
            val rxStore = fragment.rxStore
            if (rxStore is FragmentStore) {
                //rxStore关联Fragment生命周期
                fragment.lifecycle.addObserver(rxStore)
            }
        }
    }
}
